### PR TITLE
Remove phpunit from require-dev

### PR DESCRIPTION
**Pull request by harikt**
so we can test on each php version separately.

Changed scrutinizer config, and continuous integration file. Thanks @koriym .

### CODE REVIEW

File: .scrutinizer.yml [hunk 1]
```diff
@@ -1,10 +1,8 @@
...
-    paths: ["src/*"]
-tools:
-    external_code_coverage: true
-    php_code_coverage: true
-    php_sim: true
-    php_mess_detector: true
-    php_pdepend: true
-    php_analyzer: true
-    php_cpd: true
+    paths: ["src/*"]
```
Comment (koriym) [inline .scrutinizer.yml]:
```suggestion
    paths: ["src/*"]

```

File: composer.json [hunk 2]
```diff
@@ -20,7 +20,7 @@
         }
     ],
     "require": {
-        "php": "^5.6|^7.0"
+        "php": "^5.6|^7.0|^8.0"
```
Comment (koriym) [inline composer.json]:
```suggestion
        "php": "^5.6 || ^7.0 || ^8.0"
```
Single pipe works but double pipe seems more appropriate.
See https://github.com/composer/composer/issues/6755#issuecomment-337588505

**Review by koriym — COMMENTED** [review]
Just small note..

**Comment by koriym** [comment]
It's hard to cover php 5.6 to 8.1 in single version of phpunit. 5.x-8.0 or 7.2-8.1 would be the choices.

**Comment by koriym** [comment]
I've tried cover PHP 5.6-8.1 support in single version of phpunit here. 
https://github.com/auraphp/Aura.Sql/issues/201#issue-1052208268

the result
https://github.com/koriym/Aura.Sql/runs/4196010089?check_suite_focus=true

**Comment by harikt** [comment]
Hi @koriym ,

I agree we cannot support phpunit to all versions. What I was currently doing was to skip 5.6 unit tests. We are not making any code changes. And only trying to make it move forward to test on latest releases. So that people who are using it will not get stuck.

For example, cakephp was using Aura.Intl previsouly. Now I can see no dependencies, may be they moved away. So for people who are still using older cakephp have difficulty to upgrade to 8.x and will get stuck. So releasing 3.x seems to be the best than 4.x . 

ie why I am pushing 3.x currently. We can probably drop 4.x, and move to 5.x that will be more better also for this package.

**Review by koriym — APPROVED** [review]

**Comment by harikt** [comment]
@koriym I missed to see your experimental message.

That looks wonderful and great. Is there any reason you didn't push it further?

Else we can probably bring it here.

**Comment by koriym** [comment]
@harikt I'm OK with it. But not sure how other people feel that a bit hacky code. (I did not have a response.)

**Comment by koriym** [comment]
@harikt If you think that polyfill is OK,I don't disagree.

**Comment by harikt** [comment]
@koriym I actually tried something like that, but the void and all stuff was messing. So thought it is not good to waste time, ie why moved away ;-) .

**Comment by koriym** [comment]
@harikt If we want to keep CI from PHP 5.6 to the latest 8.1, It's only choice?

**Comment by harikt** [comment]
Is there anything better than that? I am not aware of any.

Looks like there are https://packagist.org/packages/yoast/phpunit-polyfills

**Comment by koriym** [comment]
@harikt Oh forgotten. yoast/phpunit-polyfills maybe better choice.

**Comment by harikt** [comment]
@koriym looks good for merging to 3.x .

**Comment by koriym** [comment]
@harikt New question. Do we not need the 4.x branch? And the other packages also?

**Comment by koriym** [comment]
We have already released for 4.x and 5.x but only for Aura.Sql. Should Aura.Session and Aura.Router also support 5.6 to 8.1 as this package does with this 3.x?

**Comment by harikt** [comment]
@koriym I have created an issue over https://github.com/auraphp/bin/issues/13 . How about we discuss this there more on it.

If we can support php versions from 5.6 to any versions we should consider using 3.x this is my take.
And newer ones can go with 8.1, so people can always change their dependency and we have less work to maintain.

We should bring all peoples attention to discuss this.

**Comment by koriym** [comment]
Ok, let's discuss it there!

**Pull request merged**